qiskit_experiments/library/randomized_benchmarking/create_clifford_map.py

```diff
@@ -0,0 +1,194 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+This is a script used to create the data in clifford_data.py.
+Every Clifford is represented by a number. We store a list of the compositions of Cliffords represented
+as numbers. For example, if Clifford1.compose(Clifford2) == Clifford3, then we conceptually,
+we store {(1, 2) : 3}. we don't actually store the map, but only the results of the compose in an array.
+This is more efficient in performance. The result is found using the indices of the input Cliffords.
+Similarly, we store for each number representing a Clifford, the number representing the
+inverse Clifford.
+For compose, we don't actually store the full compose table of all-cliffords X all-cliffords.
+Instead, we define an array of single-gate-cliffords. This comprises all Cliffords that consist
+of a single gate. There are 8 such Cliffords for 1-qubit, and 20 such Cliffords for 2-qubits.
+It is sufficient to store the compose table of all-cliffords X single-gate-cliffords,
+since for every Clifford on the right hand side, we can break it down into single gate Cliffords,
+and do the composition one at a time. This greatly reduces the storage space for the array of
+composition results (from O(n^2) to O(n)), where n is the number of Cliffords.
+"""
+import itertools
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import (
+    IGate,
+    HGate,
+    SXdgGate,
+    SGate,
+    XGate,
+    SXGate,
+    YGate,
+    ZGate,
+    SdgGate,
+    CXGate,
+)
+from qiskit.quantum_info.operators.symplectic import Clifford
+from qiskit_experiments.library.randomized_benchmarking.clifford_utils import CliffordUtils
+
+gate_list_1q = [
+    IGate(),
+    HGate(),
+    SXdgGate(),
+    SGate(),
+    XGate(),
+    SXGate(),
+    YGate(),
+    ZGate(),
+    SdgGate(),
+]
+num_to_cliff_1q = {}
+cliff_to_num_1q = {}
+num_to_cliff_2q = {}
+cliff_to_num_2q = {}
+
+
+def gen_nums_single_gate_cliffs_1q():
+    """
+    We generate an array mapping numbers to Cliffords and the reverse array.
+    Based on this array, we build a mapping from every single-gate-clifford to its number.
+    The mapping actually looks like {(gate, '[0]'): num}, where [0] represents qubit 0.
+    The qubit is added to be consistent with the format for 2 qubits.
+    """
+    for i in range(CliffordUtils.NUM_CLIFFORD_1_QUBIT):
+        cliff = CliffordUtils.clifford_1_qubit(i)
+        num_to_cliff_1q[i] = cliff
+        cliff_to_num_1q[cliff.__repr__()] = i
```

Comment:
can also do `cliff_to_num_1q[repr(cliff)] = i` (here and in many other places)

Reply:
Indeed better. Done.